docs/contributing-to-airbyte/tutorials/building-a-python-destination.md

```diff
@@ -0,0 +1,193 @@
+* Step 5: Implement `write` to write data to the destination
+* Step 6: Set up Acceptance Tests
+* Step 7: Write unit tests or integration tests
+* Step 8: Update the docs \(in `docs/integrations/destinations/<destination-name>.md`\)
+
+{% hint style="info" %}
+If you need help with any step of the process, feel free to submit a PR with your progress and any questions you have, or ask us on [slack](https://slack.airbyte.io). Also reference the KvDB python destination implementation if you want to see an example of a working destination. 
+{% endhint %}
+
+## Explaining Each Step
+
+### Step 1: Create the destination using the template
+
+Airbyte provides a code generator which bootstraps the scaffolding for our connector.
+
+```bash
+$ cd airbyte-integrations/connector-templates/generator # assumes you are starting from the root of the Airbyte project.
+$ ./generate.sh
+```
+
+Select the `Python Destination` template and then input the name of your connector. We'll refer to the destination as `destination-<name>` in this tutorial, but you should replace `<name>` with the actual name you used for your connector e.g: `redis` or `google-sheets`. 
+
+### Step 2: Setup the dev environment
+
+Setup your Python virtual environment: 
+
+```bash
+cd airbyte-integrations/connectors/destination-<name>
+# Create a virtual environment in the .venv directory
+python -m venv .venv 
+# activate the virtualenv
+source .venv/bin/activate 
+pip install -r requirements.txt
+```
+This step sets up the initial python environment. **All** subsequent `python` or `pip` commands assume you have activated your virtual environment.
+
+If you want your IDE to auto complete and resolve dependencies properly, point it at the python binary in `airbyte-integrations/connectors/destination-<name>/.venv/bin/python`. Also anytime you change the dependencies in the `setup.py` make sure to re-run the build command. The build system will handle installing all dependencies in the `setup.py` into the virtual environment.
+
+Let's quickly get a few housekeeping items out of the way.
+
+#### Dependencies
+
+Python dependencies for your destination should be declared in `airbyte-integrations/connectors/destination-<name>/setup.py` in the `install_requires` field. You might notice that a couple of Airbyte dependencies are already declared there (mainly the Airbyte CDK and potentially some testing libraries or helpers). Keep those as they will be useful during development. 
+
+You may notice that there is a `requirements.txt` in your destination's directory as well. Do not touch this. It is autogenerated and used to install local Airbyte dependencies which are not published to PyPI. All your dependencies should be declared in `setup.py`.
+
+#### Iterating on your implementation
+
+Pretty much all it takes to create a destination is to implement the `Destination` interface. Let's briefly recap the three methods implemented by a Destination: 
```

Comment:
I mentioned this somewhere else - is there a good reason to keep the requirements.txt? if this is only references local dependencies, can we get rid of it?

Reply:
The only reason for sources to keep `requirements.txt` was SAT being a part of mono repo. Since we don't have SAT for destinations yet I don't see any good reason to keep it.

Reply:
removed it, but there is non trivial chance it comes back so keeping the section saying "if there is a requirements.txt don't touch it"